cornflow/schemas/schema_manager.py

```diff
@@ -0,0 +1,368 @@
+    
+        self.flask_schemas = result_dict['DataSchema']
+        return self.flask_schemas
+    
+    def jsonschema_to_flask(self):
+        """
+        Create flask marshmallow schemas from the jsonschema
+
+        :return: a dict containing the flask marshmallow schemas
+        """
+        self.jsonschema_to_dict()
+        self.dict_to_flask()
+        return self.flask_schemas
+        
+    def load_schema(self, path):
+        """
+        Load a json schema into self.jsonschema.
+
+        :param path: the path to the json file containing the schema.json
+
+        return the jsonschema
+        """
+        self.jsonschema = self.load_json(path)
+        return self.jsonschema
+
+    def print_schema_dict(self, path):
+        """
+        Print the schema_dict in a json file.
+
+        :param path: the path where to save the dict.format
+
+        :return: nothing
+        """
+        self.save_json(self.schema_dict, path)
+    
+    def draft_schema_from(self, path, save_path=None):
+        """
+        Create a draft jsonschema from a json file of data.
+        
+        :param path: path to the json file.
+        :param save_path: path where to save the generated schema.
+        
+        :return: the generated schema.
+        """
+        
+        file = self.load_json(path)
+        
+        builder = SchemaBuilder()
+        builder.add_schema({"type": "object", "properties": {}})
+        builder.add_object(file)
+        
+        self.draft_schema = builder.to_json()
+        if save_path is not None:
+            self.save_json(self.draft_schema, save_path)
+        return self.draft_schema
+    
+    """
+    Functions used to translate jsonschema to schema_dict
+    """
+    def _create_data_schema(self, item):
+        """
+        Add a schema to self.schema_dict["DataSchema"]
+
+        :param item: (key, value) of a dict. The key cotain the name of the schema and the value containt its content.
+
+        return the schema dict.
+        """
+    
+        name = item[0]
+        content = item[1]
+    
+        schema = [dict(name=name,
+                      type=self._get_schema_name(name),
+                      many=("type" in content and content["type"] == "array"),
+                      required=True)]
+        self.schema_dict["DataSchema"] +=schema
+    
+        return schema
+    
+    def _get_element_dict(self, item, required_list=None):
+        """
+        Parse an item (key, value) from the jsonschema and return the corresponding dict.
+        
+        :param item: An item from the jsonschema (key, value)
+        :param required_list: A list of names corresponding to the required fields in the parent object
+        
+        :return A dict element for a schema_dict.
+        """
+        if required_list is None:
+            required_list = []
+        
+        name = item[0]
+        content = item[1]
+        if "type" not in content:
+            if "$ref" in content:
```

Comment:
what happens when this inside if is false? do we get an error?

Reply:
good question, I do not know what we should do if no type is given:
- raise an error here?
- make it a string by default?
- raise an error when transforming it into a flask schema?

Reply:
I have no idea what the code does actually, I just noticed the ifs. If it's wrong (unexpected): we should raise an error saying "incorrect jsonschema" and that the object `name` is missing a type. I've just checked that "raise TypeError()" exists so you could use that haha.